packages/circuit-ui/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,277 @@
+  /**
+   * The ref to the HTML DOM element, can be a button or an anchor.
+   */
+  ref?: React.Ref<HTMLButtonElement & HTMLAnchorElement>;
+}
+
+type LinkElProps = Omit<
+  HTMLProps<HTMLAnchorElement>,
+  'size' | 'type' | 'onClick'
+>;
+type ButtonElProps = Omit<
+  HTMLProps<HTMLButtonElement>,
+  'size' | 'type' | 'onClick'
+>;
+
+export type PopoverItemProps = BaseProps & LinkElProps & ButtonElProps;
+type PopoverItemWrapperProps = LinkElProps & ButtonElProps;
+
+const itemWrapperStyles = () => css`
+  label: popover-item;
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+  width: 100%;
+`;
+
+const PopoverItemWrapper = styled('button')<PopoverItemWrapperProps>(
+  listItem,
+  itemWrapperStyles,
+  typography('one'),
+);
+
+const iconStyles = (theme: Theme) => css`
+  label: popover__icon;
+  margin-right: ${theme.spacings.byte};
+`;
+
+export const PopoverItem = ({
+  children,
+  icon: Icon,
+  onClick,
+  tracking,
+  ...props
+}: PopoverItemProps): JSX.Element => {
+  const components = useComponents();
+
+  // Need to typecast here because the PopoverItemWrapper expects a button-like
+  // component for its `as` prop. It's safe to ignore that constraint here.
+  /* eslint-disable @typescript-eslint/no-unsafe-assignment */
+  const Link = components.Link as any;
+
+  const handleClick = useClickHandler<MouseEvent | KeyboardEvent>(
+    onClick,
+    tracking,
+    'popover-item',
+  );
+
+  return (
+    <PopoverItemWrapper
+      as={props.href ? Link : 'button'}
+      onClick={handleClick}
+      {...props}
+    >
+      {Icon && <Icon css={iconStyles} />}
+      {children}
+    </PopoverItemWrapper>
+  );
+};
+
+const wrapperStyles = ({ theme }: StyleProps) => css`
+  label: popover;
+  padding: ${theme.spacings.byte} 0px;
+  border: 1px solid ${theme.colors.n200};
+  box-sizing: border-box;
+  box-shadow: 0px 3px 8px rgba(0, 0, 0, 0.2);
+  border-radius: 8px;
+  background-color: ${theme.colors.white};
+
+  ${theme.mq.untilKilo} {
+    border-bottom-right-radius: 0;
+    border-bottom-left-radius: 0;
+  }
+`;
+
+const PopoverWrapper = styled('div')(wrapperStyles);
+
+const dividerStyles = (theme: Theme) => css`
+  margin: ${theme.spacings.byte} ${theme.spacings.mega};
+  width: calc(100% - ${theme.spacings.mega}*2);
+`;
+
+const Overlay = styled.div(
+  ({ theme }: StyleProps) => css`
+    ${theme.mq.untilKilo} {
+      position: fixed;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      background-color: ${theme.colors.overlay};
+    }
+  `,
+);
+
+type Divider = { type: 'divider' };
+type Action = PopoverItemProps | Divider;
+
+function isDivider(action: Action): action is Divider {
+  return 'type' in action && action.type === 'divider';
+}
+
+export interface PopoverProps {
+  isOpen: boolean;
+  actions: Action[];
+  referenceElement: RefObject<HTMLElement>;
+  placement?: Placement;
+  fallbackPlacements?: Placement[];
+  onClose: (event: Event) => void;
+}
```

Comment:
It would be great to add prop descriptions here, these are the ones that developers will interact with 🙂